ansible_runner/utils/capacity.py

```diff
@@ -22,16 +27,27 @@ def get_mem_in_bytes():
 
 
 def get_uuid():
-    try:
-        with open('/var/lib/dbus/machine-id') as f:
-            uuid = f.read()
-        return uuid.strip()
-    except FileNotFoundError:
-        try:
-            with open('/etc/machine-id') as f:
-                uuid = f.read()
-                return uuid.strip()
-        except FileNotFoundError:
-            error = ("Could not find /var/lib/dbus/machine-id or "
-                     "/etc/machine-id files, UUID undiscoverable.")
-            return error
+    uuid_directory = Path('/etc/ansible/facts.d')
+    if uuid_directory.exists():
+        # Read the contents of the uuid.txt file if it already exists
+        uuid_file_path = os.path.join(uuid_directory[0], 'uuid.txt')
+        with open(uuid_file_path) as f:
+            saved_uuid = f.read()
+        return saved_uuid
+    else:
+        # Generate a new UUID if no uuid.txt file is found
+        newly_generated_uuid = generate_uuid()
+        return newly_generated_uuid
+
+
+def generate_uuid():
```

Comment:
Can this function get marked private? I only see it being called internally in this module.